impeller/typographer/backends/skia/text_render_context_skia.cc

```diff
@@ -81,8 +73,9 @@ static size_t PairsFitInAtlasOfSize(
   glyph_positions.clear();
   glyph_positions.reserve(pairs.size());
 
-  for (size_t i = 0; i < pairs.size(); i++) {
-    const auto& pair = pairs[i];
+  size_t i = 0;
+  for (auto it = pairs.begin(); it != pairs.end(); ++i, ++it) {
+    const auto& pair = *it;
```

Comment:
I don't really love this. Does it make a significant improvement on the generated code?

Reply:
It was required because you can't index into `std::unordered_set`.  We could use a foreach loop but it isn't really cleaner imo.

Previously we were copying all of the data in the `std::unordered_set` to make a `std::vector` just so we could cleanly index into the data.  But since the order of iteration of an `std::unordered_set` is deterministic, we don't really need to copy the data just to index into it.

Reply:
makes sense